added some things from testdm

Skooma-Breath opened the pull request:
made a new file with the map data and also added the brown and blue uniforms from testdm.

also merged the matchesspawnblueplate and brownplate functions into a single matchesSpawnPlayers funciton, seemed to make sense at the time.

Skooma-Breath committed:
.

Skooma-Breath committed:
fixed conflict

Skooma-Breath committed:
added maps and mapdata in a new file and also the blue and brown uniforms from testdm

Skooma-Breath committed:
removed comments

Skooma-Breath committed:
added map file

Skooma-Breath committed:
removed old comments

Skooma-Breath committed:
Update plateWars.lua

Nkfree commented:
For map data a json would be more useful because of how easy it is to write it - assume you want to add a new spawnpoint in-game, you can just have a function and do it on the fly, instead of having to add the data manually or in the case of modifying .lua file, having to code-in how to format it.
```
plateWarsMaps.aldruhn.teamSpawnLocations = {}
plateWarsMaps.aldruhn.teamSpawnLocations[1] = {
	{"-2, 7", -13092, 57668, 2593, 2.39},
	{"-2, 7", -12208, 57532, 2594, 3.11},
	{"-2, 7", -11638, 57629, 2593, -2.78},
	{"-2, 7", -11352, 56683, 2593, -2.26},
	{"-2, 7", -11798, 56952, 2586, -2.73},
	{"-2, 7", -12508, 56971, 2583, 3.11},
	{"-2, 7", -12970, 56871, 2588, 2.67}
```
- it seems cleaner to move the cell string out of the coords table and make it a key like:
```
["-2, -7"] = {
 {coords1},
 {coords2},
 ...
}
```
- why not use bluePlates and brownPlates tables inside teamSpawnLocations, it's easier to refer to the specific team by the named keys than by index: 
_map.teamSpawnLocations.bluePlates_ - I know where I am
_map.teamSpawnLocations[1]_ - What was it again? Was it blue plates or brown plates? Let me take a look.

You then don't need the teamIndex at all
```
plateWars.teams.baseData = {
    bluePlatesPids = {},
    brownPlatesPids = {}
    brownPlatesPids = {},
    bluePlatesIndex = 1,
    brownPlatesIndex = 2
}
```

-----
```
function plateWars.LoadPlayerItems(pid)
          Players[pid]:Save(pid)
	  Players[pid]:LoadInventory(pid)
	  Players[pid]:LoadEquipment(pid)
end
```

- LoadPlayerItems: none of the functions called inside takes the argument _pid_, so it's Players[pid]:Save() because the correct player is defined using Players[pid], it doesn't crash because the argument is unused thus discarded, yet it is not correct, same for the rest of the functions.
- using Players[pid]:Save() is rather expensive function because it also tries to pretty format the json to make it human readable and priority of keys can be specified, so for example player.inventory is above equipment in the result json file. Instead of that use Players[pid]:QuicksaveToDrive() - this by default doesn't format anything, nor doesn't prioritize any keys which makes it faster.

Skooma-Breath commented:
sounds good to me

Skooma-Breath committed:
changed mapfile to json

changed spawnlocation keys from numbers to team names

changed save() to QuicksaveToDrive()

Skooma-Breath committed:
Merge branch 'main' of https://github.com/Skooma-Breath/tes3mp-platewars into main

Skooma-Breath committed:
Update plateWars.lua

Skooma-Breath committed:
Update plateWars.lua

Skooma-Breath committed:
Update plateWars.lua

Skooma-Breath commented:
did you mean like this?

```
plateWarsMaps.aldruhn.teamSpawnLocations = {}
plateWarsMaps.aldruhn.teamSpawnLocations["bluePlates"] = {
	{"-2, 7", -13092, 57668, 2593, 2.39},
	{"-2, 7", -12208, 57532, 2594, 3.11},
	{"-2, 7", -11638, 57629, 2593, -2.78},
	{"-2, 7", -11352, 56683, 2593, -2.26},
	{"-2, 7", -11798, 56952, 2586, -2.73},
	{"-2, 7", -12508, 56971, 2583, 3.11},
	{"-2, 7", -12970, 56871, 2588, 2.67}
}
plateWarsMaps.aldruhn.teamSpawnLocations["brownPlates"] = {
	{"-2, 6", -15047, 51823, 2065, 0.72},
	{"-2, 6", -14996, 52449, 2065, 1.06},
	{"-2, 6", -14920, 52999, 2176, 1.87},
	{"-2, 6", -14301, 52838, 2167, 1.54},
	{"-2, 6", -13892, 52524, 2106, 1.21},
	{"-2, 6", -13124, 51895, 2111, -0.09},
	{"-2, 6", -13763, 51485, 2077, 0.45}
}
```

or like this
```
plateWarsMaps.aldruhn.teamSpawnLocations = {}
plateWarsMaps.aldruhn.teamSpawnLocations["-2, 7"] = {
	{-13092, 57668, 2593, 2.39},
	{-12208, 57532, 2594, 3.11},
	{-11638, 57629, 2593, -2.78},
	{-11352, 56683, 2593, -2.26},
	{-11798, 56952, 2586, -2.73},
	{-12508, 56971, 2583, 3.11},
	{-12970, 56871, 2588, 2.67}
}
plateWarsMaps.aldruhn.teamSpawnLocations["-2, 6"] = {
	{-15047, 51823, 2065, 0.72},
	{-14996, 52449, 2065, 1.06},
	{-14920, 52999, 2176, 1.87},
	{-14301, 52838, 2167, 1.54},
	{-13892, 52524, 2106, 1.21},
	{-13124, 51895, 2111, -0.09},
	{-13763, 51485, 2077, 0.45}
}
```

Skooma-Breath committed:
remove head

Skooma-Breath committed:
Merge branch 'main' of https://github.com/Skooma-Breath/tes3mp-platewars into main

Nkfree commented:
I meant it like this:

```plateWarsMaps.aldruhn.teamSpawnLocations = {}
plateWarsMaps.aldruhn.teamSpawnLocations.bluePlates = {}
plateWarsMaps.aldruhn.teamSpawnLocations.bluePlates["-2, 7"] = {
	{-13092, 57668, 2593, 2.39},
	{-12208, 57532, 2594, 3.11},
	{-11638, 57629, 2593, -2.78},
	{-11352, 56683, 2593, -2.26},
	{-11798, 56952, 2586, -2.73},
	{-12508, 56971, 2583, 3.11},
	{-12970, 56871, 2588, 2.67}
}
```
same for brownPlates.
Notice that _teamSpawnLocations.bluePlates_ is equivalent to _teamSpawnLocations["bluePlates"]_, it's just easier to write.

Anyway, we could first discuss what you guys think is a better way of structuring a spawnLocation. Should it be:
```plateWarsMaps.aldruhn.teamSpawnLocations.bluePlates["-2, 7"]``` 
or 
```plateWarsMaps.aldruhn.teamSpawnLocations["-2, 7"].bluePlates```, i.e. team keys first and cell keys inside or cell key first and team keys inside?

Skooma-Breath commented:
Feel free to figure out whatever is best, makes no difference to me.

Skooma-Breath committed:
changed structure of json and updated matchesspawnplayers

Skooma-Breath commented:
> I meant it like this:
> 
> ```
> plateWarsMaps.aldruhn.teamSpawnLocations.bluePlates = {}
> plateWarsMaps.aldruhn.teamSpawnLocations.bluePlates["-2, 7"] = {
> 	{-13092, 57668, 2593, 2.39},
> 	{-12208, 57532, 2594, 3.11},
> 	{-11638, 57629, 2593, -2.78},
> 	{-11352, 56683, 2593, -2.26},
> 	{-11798, 56952, 2586, -2.73},
> 	{-12508, 56971, 2583, 3.11},
> 	{-12970, 56871, 2588, 2.67}
> }
> ```
> 
> same for brownPlates. Notice that _teamSpawnLocations.bluePlates_ is equivalent to _teamSpawnLocations["bluePlates"]_, it's just easier to write.
> 
> Anyway, we could first discuss what you guys think is a better way of structuring a spawnLocation. Should it be: `plateWarsMaps.aldruhn.teamSpawnLocations.bluePlates["-2, 7"]` or `plateWarsMaps.aldruhn.teamSpawnLocations["-2, 7"].bluePlates`, i.e. team keys first and cell keys inside or cell key first and team keys inside?

i've updated the structure to your first suggestion here

Nkfree commented:
The json maps file looks amazing!

I'm sorry, I decided to do dive into some other things (as expected), so I don't know when I will post an update but surely the json is something to merge in.

The rest of the code should be edited to support multiple matches going on. If one is currently taking place, it's fine but the benefit is that we won't have to redo it later. To support multiple matches means that any function that should only operate with match specific data needs to have the match or matchId (from which you can get the match table) as a parameter and anything related to match specific data inside the function needs to reflect on the passed argument, can't express myself properly, so I'll try and provide a simple example:

**Doesn't support multiple matches:**
```
function plateWars.teamJoinBluePlates(pid, match)
    plateWars.teamJoin(pid, plateWars.teams.baseData.bluePlatesPids)
end
```

**Does support multiple matches:**
```
function plateWars.teamJoinBluePlates(pid, match)
    plateWars.teamJoin(pid, match.data.teams.bluePlatesPids)
end
```

or

```
function plateWars.teamJoinBluePlates(pid, matchId)
    local match = plateWars.getMatch(matchId)
    plateWars.teamJoin(pid, match.data.teams.bluePlatesPids)
end
```

Skooma-Breath committed:
changed a bunch of functions to use the match data from the instances table

Skooma-Breath committed:
^

Skooma-Breath committed:
^

Skooma-Breath committed:
fixed stuff

Skooma-Breath commented:
hmmm are we gonna need a function like matchesGetCurrentMatch(pid) that checks which match instance a pid is from? idk i'll let you figure out where you want to go with the multiple matches thing
